### ISSUE TITLE
Https scheme not working - InvalidAlgorithmParameterException - ECDHE

### ISSUE THREAD

**Issue by brommer** [uken/fluent-plugin-elasticsearch#258]
#### Problem
Fluentd cannot connect to Elasticsearch using HTTPS scheme. Elasticsearch is throwing following error: 
`
Caused by: java.security.ProviderException: java.security.InvalidAlgorithmParameterException
        at sun.security.ec.ECKeyPairGenerator.generateKeyPair(ECKeyPairGenerator.java:147) ~[?:?]
        at java.security.KeyPairGenerator$Delegate.generateKeyPair(KeyPairGenerator.java:703) ~[?:1.8.0_121]
        at sun.security.ssl.ECDHCrypt.<init>(ECDHCrypt.java:65) ~[?:?]
        at sun.security.ssl.ServerHandshaker.setupEphemeralECDHKeys(ServerHandshaker.java:1431) ~[?:?]
        at sun.security.ssl.ServerHandshaker.trySetCipherSuite(ServerHandshaker.java:1229) ~[?:?]
        at sun.security.ssl.ServerHandshaker.chooseCipherSuite(ServerHandshaker.java:1026) ~[?:?]
        at sun.security.ssl.ServerHandshaker.clientHello(ServerHandshaker.java:741) ~[?:?]
        at sun.security.ssl.ServerHandshaker.processMessage(ServerHandshaker.java:224) ~[?:?]
        at sun.security.ssl.Handshaker.processLoop(Handshaker.java:1026) ~[?:?]
        at sun.security.ssl.Handshaker$1.run(Handshaker.java:966) ~[?:?]
        at sun.security.ssl.Handshaker$1.run(Handshaker.java:963) ~[?:?]
        at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0_121]
        at sun.security.ssl.Handshaker$DelegatedTask.run(Handshaker.java:1416) ~[?:?]
        at io.netty.handler.ssl.SslHandler.runDelegatedTasks(SslHandler.java:1167) ~[?:?]
        at io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1080) ~[?:?]
        at io.netty.handler.ssl.SslHandler.decode(SslHandler.java:950) ~[?:?]
        at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:411) ~[?:?]
        ... 15 more
`

When changing the default Elasticsearch cipher suites to "TLS_RSA_WITH_AES_128_CBC_SHA256, TLS_RSA_WITH_AES_128_CBC_SHA" only (not using any of the ECDHE defaults TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256, TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256, TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA, TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA - see https://www.elastic.co/guide/en/x-pack/current/security-settings.html#ssl-tls-settings), I am able to establish a connection.

I think this must be working with default cipher suite settings as well. 

#### Steps to replicate
- Create a fluentd Docker image:
```
FROM fluent/fluentd:v0.14.15
USER root
RUN gem install fluent-plugin-elasticsearch -v 1.9.3
USER fluent
```

- Use https scheme in fluentd. 

- Check Elasticsearch logs

**Comment by cosmo0920** [comment]
I think that this is SSL/TLS version mismatch issue.
Could you try https://github.com/uken/fluent-plugin-elasticsearch/pull/299 patch?

**Comment by cosmo0920** [comment]
I've released v0.10.0 and v2.0.0.rc.5.
Could you try them?

**Comment by cosmo0920** [comment]
No response. Closing.
If you encountered an issue which is related to this, please reopen.
